.changeset/few-mirrors-reflect.md

```diff
@@ -0,0 +1,23 @@
+---
+"@latticexyz/cli": major
+"@latticexyz/store": major
+"@latticexyz/world": major
+"create-mud": patch
+---
+
+All `Store` methods now require the table's value schema to be passed in as an argument instead of loading it from storage.
+This decreases gas cost and removes circular dependencies of the Schema table (where it was not possible to write to the Schema table before the Schema table was registered).
+
+```diff
+  function setRecord(
+    bytes32 table,
+    bytes32[] calldata key,
+    bytes calldata data,
++   Schema valueSchema
```

Comment:
Does the argument ordering here make sense? I usually order args by specificity and would kind of expect `valueSchema` to be further ahead in the list of arguments (before or after `table` perhaps).

Unless it's an optional arg, then it being at the end makes a bit more sense.

Reply:
i agree with ordering by specificity but would defer this refactor until after this current stack of PRs is merged since they touch lots of the same places in the code and changing it here would likely lead to massive merge conflicts in the upstream branches